From 9d265dc548c5d1e3df9129a15865e70b1597451d Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Wed, 2 Mar 2016 23:51:51 -0600
Subject: [PATCH] Don't create Region File's when checking if chunk exists

Plugins like Dynmap can end up creating tons of emtpy Region Files
when using chunkExists.

diff --git a/src/main/java/net/minecraft/server/RegionFileCache.java b/src/main/java/net/minecraft/server/RegionFileCache.java
index 6ec72689c..0163a4e59 100644
--- a/src/main/java/net/minecraft/server/RegionFileCache.java
+++ b/src/main/java/net/minecraft/server/RegionFileCache.java
@@ -12,7 +12,13 @@ public class RegionFileCache {
 
     public static final Map<File, RegionFile> a = Maps.newHashMap(); // Spigot - private -> public
 
+    // Paper start
     public static synchronized RegionFile a(File file, int i, int j) {
+        return a(file, i, j, true);
+    }
+
+    public static synchronized RegionFile a(File file, int i, int j, boolean create) {
+        // Paper end
         File file1 = new File(file, "region");
         File file2 = new File(file1, "r." + (i >> 5) + "." + (j >> 5) + ".mca");
         RegionFile regionfile = (RegionFile) RegionFileCache.a.get(file2);
@@ -20,6 +26,7 @@ public class RegionFileCache {
         if (regionfile != null) {
             return regionfile;
         } else {
+            if (!create && !file2.exists()) { return null; } // Paper
             if (!file1.exists()) {
                 file1.mkdirs();
             }
-- 
2.12.2.windows.2

